### ISSUE TITLE
Serialize Kotlin objects to dynamic JS objects

### ISSUE THREAD

**Issue by nlochschmidt** [Kotlin/kotlinx.serialization#116]
I saw that it's possible to parse `dynamic` objects in Javascript to data classes using koltinx.serialization https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/runtime_usage.md#dynamic-object-parser-js-only

I think it would be a useful addition to be able to serialize directly into a `dynamic` object as well. 
At the moment I can only do this (and go through an intermediate `String` instance):

```kotlin
val jsonString = kotlinx.serialization.json.JSON.stringify(kotlinData)
val dataAsPureJs = kotlin.js.JSON.parse(jsonString)
```

However it would be nice to be able to do this without the intermediate representation.

**Comment by fkrauthan** [comment]
Any plans for thought when this feature might be added to this library? This would be a huge win for frameworks where you have to work a lot with json objects to persist and restore data.

**Comment by ankushg** [comment]
Is this still not possible at this time?

**Comment by sandwwraith** [comment]
Nope :(

**Comment by ankushg** [comment]
Would it be possible to create a JS-platform-only `AbstractSerialFormat` that spits out `dynamic`?

**Comment by sandwwraith** [comment]
Yep, that's how it should be done. Similar to `DynamicObjectParser`.

**Comment by exaV** [comment]
Would a PR be welcome?

**Comment by sandwwraith** [comment]
Fixed in #628
